### PR TITLE
Fix function calls in configs

### PR DESCRIPTION
**Pull request by BenignBeppe**
Fixed a bug where `config_reader` couldn't load config files containing
function calls. The call for a function used to set a config variable now
occurs the first time it is used. The syntax for the config files was
changed to match the new implementation.

Task: https://phabricator.wikimedia.org/T157933

### CODE REVIEW

File: config_reader.py [hunk 1]
```diff
@@ -25,4 +25,13 @@ def get(variable):
         raise Exception(
             "Variable '{}' not found in config file.".format(variable)
         )
+    value = getattr(config, variable)
+    # If the value is a tuple where the first item is a function, call
+    # the function with the remaining items as arguments. The value is
+    # set to the return value, so that the function doesn't need to be
+    # called if the variable is read again.
+    if type(value) == tuple and callable(value[0]):
```
Comment (lokal-profil) [inline config_reader.py]: I believe the first comparison should be `isinstance(value, tuple)`

File: scripts/teahouse/send_th_invites.py [hunk 2]
```diff
@@ -152,9 +152,11 @@ def inviteGuests(prof, inviter):
...
     )
-    logging.info("Reading configuration file: '{}'.".format(args.config_file))
-    config_reader.load_config_module(args.config_file)
     try:
+        logging.info("Reading configuration file: '{}'.".format(
+            args.config_file)
+        )
+        config_reader.load_config_module(args.config_file)
         send_invitations()
     except:
```
Comment (lokal-profil) [inline scripts/teahouse/send_th_invites.py:161]: Do we actually want to also log the exception which caused this?

File: hb_config.sample.py [hunk 3]
```diff
@@ -37,8 +37,11 @@
 max_user_age = 2
 
 # The minimum number of edits that a user must have made to qualify
-# for invitation.
-min_edits = page_reader.read_json_value(
+# for invitation. Note that the syntax for using the return value of a
```
Comment (lokal-profil) [inline hb_config.sample.py:40]: Add something here (ore elsewhere in the config) about it not being possible to use keyword arguments.

**Review by lokal-profil — CHANGES_REQUESTED** [review]
Looks good overall

**Comment by lokal-profil** [comment]
I also added the task to the PR description

File: config_reader.py (hunk 1 above)
Comment (BenignBeppe) [reply config_reader.py]: Fixed. I don't know what the difference is, but this looks nicer.

File: hb_config.sample.py (hunk 3 above)
Comment (BenignBeppe) [reply hb_config.sample.py:40]: Fixed.

File: scripts/teahouse/send_th_invites.py (hunk 2 above)
Comment (BenignBeppe) [reply scripts/teahouse/send_th_invites.py:161]: [`logging.exception`](https://docs.python.org/2/library/logging.html?highlight=logging#logging.Logger.exception) logs the original exception.

**Comment by lokal-profil** [comment]
+2 from mobile

**Comment by lokal-profil** [comment]
We're you by any chance logged in on labs when committing the code ;)

**Pull request merged**